eZ/Publish/Core/Persistence/Legacy/User/Role/Gateway/DoctrineDatabase.php

```diff
@@ -57,492 +57,252 @@ public function createRole(Role $role)
             $roleOriginalId = Role::STATUS_DEFINED;
         }
 
-        $query = $this->handler->createInsertQuery();
+        $query = $this->connection->createQueryBuilder();
         $query
-            ->insertInto($this->handler->quoteTable('ezrole'))
-            ->set(
-                $this->handler->quoteColumn('id'),
-                $this->handler->getAutoIncrementValue('ezrole', 'id')
-            )->set(
-                $this->handler->quoteColumn('is_new'),
-                0
-            )->set(
-                $this->handler->quoteColumn('name'),
-                $query->bindValue($role->identifier)
-            )->set(
-                $this->handler->quoteColumn('value'),
-                0
-            )->set(
-                // Column name "version" is misleading here as it stores originalId when creating a draft from an existing role.
-                // But hey, this is legacy! :-)
-                $this->handler->quoteColumn('version'),
-                $query->bindValue($roleOriginalId)
+            ->insert(self::ROLE_TABLE)
+            ->values(
+                [
+                    'is_new' => $query->createPositionalParameter(0, ParameterType::INTEGER),
+                    'name' => $query->createPositionalParameter(
+                        $role->identifier,
+                        ParameterType::STRING
+                    ),
+                    'value' => $query->createPositionalParameter(0, ParameterType::INTEGER),
+                    'version' => $query->createPositionalParameter(
+                    // BC: "version" stores originalId when creating a draft from an existing role
```

Comment:
Nitpick: incorrect indent

```suggestion
                        // BC: "version" stores originalId when creating a draft from an existing role
```